src/insexpand.c

```diff
@@ -681,6 +680,51 @@ ins_compl_add(
     return OK;
 }
 
+/*
+ * Add a match to the list of matches.
+ * If the given string is already in the list of completions, then return
+ * NOTDONE, otherwise add it to the list and return OK.  If there is an error,
+ * maybe because alloc() returns NULL, then FAIL is returned.
+ */
+    static int
+ins_compl_add(
+    char_u	*str,
+    int		len,
+    char_u	*fname,
+    char_u	**cptext,	    // extra text for popup menu or NULL
+    typval_T	*user_data UNUSED,  // "user_data" entry or NULL
+    int		cdir,
+    int		flags_arg,
+    int		adup)		// accept duplicate match
+{
+    ui_breakcheck();
+    return do_ins_compl_add(
+	str, len, fname, cptext, user_data, cdir, flags_arg, adup);
+}
+
+/*
+ * Add a match to the list of matches, when adding them in a very fast tight
+ * loop, such as f_complete().
+ * If the given string is already in the list of completions, then return
+ * NOTDONE, otherwise add it to the list and return OK.  If there is an error,
+ * maybe because alloc() returns NULL, then FAIL is returned.
+ */
+    static int
+ins_compl_add_fast(
```

Comment:
Instead of duplicating the function, why not add `int fast` as another parameter?

Reply:
Seemed more explicit. Less branchy fewer args the stack. Habit I guess.